src/preferences/dialog/dlgprefautodjdlg.ui

```diff
@@ -121,8 +121,8 @@
          <property name="displayFormat">
           <string>hh:mm</string>
          </property>
-         <property name="timeSpec">
-          <enum>Qt::LocalTime</enum>
+         <property name="timeZone">
+          <enum>QTimeZone::LocalTime</enum>
```

Comment:
I think we'll need to revert this for now and leave a TODO (preferably here and maybe also in `util/datetime.h` so we don't forget about it when we can remove the version checks).

Reply:
Hmm, I cannot build with `-DWARNINGS_FATAL=ON` anymore:

```cpp
In file included from src/preferences/dialog/dlgprefautodj.h:4,
                 from src/preferences/dialog/dlgprefautodj.cpp:1:
build-2.5/mixxx-lib_autogen/include/preferences/dialog/ui_dlgprefautodjdlg.h: In member function ‘void Ui_DlgPrefAutoDJDlg::setupUi(QWidget*)’:
build-2.5/mixxx-lib_autogen/include/preferences/dialog/ui_dlgprefautodjdlg.h:103:43: error: ‘void QDateTimeEdit::setTimeSpec(Qt::TimeSpec)’ is deprecated: Use setTimeZone() instead [-Werror=deprecated-declarations]
  103 |         RequeueIgnoreTimeEdit->setTimeSpec(Qt::LocalTime);
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
In file included from /usr/include/qt6/QtWidgets/QTimeEdit:1,
                 from build-2.5/mixxx-lib_autogen/include/preferences/dialog/ui_dlgprefautodjdlg.h:20:
/usr/include/qt6/QtWidgets/qdatetimeedit.h:135:10: note: declared here
  135 |     void setTimeSpec(Qt::TimeSpec spec);
      |          ^~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/mixxx-lib.dir/build.make:4738: CMakeFiles/mixxx-lib.dir/src/preferences/dialog/dlgprefautodj.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from src/preferences/dialog/dlgprefautodj.h:4,
                 from src/preferences/dialog/dlgpreferences.cpp:25:
build-2.5/mixxx-lib_autogen/include/preferences/dialog/ui_dlgprefautodjdlg.h: In member function ‘void Ui_DlgPrefAutoDJDlg::setupUi(QWidget*)’:
build-2.5/mixxx-lib_autogen/include/preferences/dialog/ui_dlgprefautodjdlg.h:103:43: error: ‘void QDateTimeEdit::setTimeSpec(Qt::TimeSpec)’ is deprecated: Use setTimeZone() instead [-Werror=deprecated-declarations]
  103 |         RequeueIgnoreTimeEdit->setTimeSpec(Qt::LocalTime);
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
In file included from /usr/include/qt6/QtWidgets/QTimeEdit:1,
                 from build-2.5/mixxx-lib_autogen/include/preferences/dialog/ui_dlgprefautodjdlg.h:20:
/usr/include/qt6/QtWidgets/qdatetimeedit.h:135:10: note: declared here
  135 |     void setTimeSpec(Qt::TimeSpec spec);
      |          ^~~~~~~~~~~
cc1plus: all warnings being treated as errors
```
Can we maybe move this from that XML into the code (so that we can use `QT_VERSION_CHECK`?

CC'ing @ronso0 as our UI expert ;-)

Reply:
Ok, this should work: https://github.com/mixxxdj/mixxx/pull/13845/commits/723f703ecf688718712e2c1d8db4cb189367206b

Let me know if this makes sense to you, than I'll squash the fixup commit.

Reply:
Makes sense to me once you can make it build.

Reply:
needs to be `QT_VERSION_CHECK(6, 7, 0)`. not 6.5.

Reply:
Can you point out why? I don't see it in the docs: https://doc.qt.io/qt-6/qdatetime.html#setTimeZone

Just that `setTimeSpec` will be removed in Qt 6.9.

Reply:
ahh https://doc.qt.io/qt-6/qdatetimeedit.html#timeZone-prop

Reply:
yeah, sorry, I was just about to post that but I got distracted :sweat_smile: